### DISCUSSION TITLE
一个未来关于 Date() 的可能改进

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#691] [Ideas]
https://tc39.es/proposal-temporal/docs/

还是在 proposal 阶段，暂时观望一下

**Comment by Arondight** [comment]
应该跟我们没关系，肯定会向前兼容的

**Reply by mark9804** [reply]
不影响 Date() ，Date() 现在也没人敢动了，新标准直接开一个 Temporal 新坑扩展 Date 的功能，例如我们计算深渊期数调用了 `moment.js` ，因为 

1. Date() 对象中没有时区信息，VPS 时区不一定是上海时间，但是计算第几期必须使用标准的上海时间，Date() 做不到
2. Date() 的时间相加减计算返回时间戳，需要自行处理成月份

Temporal 相当于 Date() + moment.js ，甚至提案者就是 moment.js 作者，换句话说就是以后能少调一个包。Temporal转正之后我打算在抽卡页面试试

https://github.com/Arondight/Adachi-BOT/blob/14c104a12c701926db27bf591bb584c4b5725b7e/src/utils/detail.js#L106-L118
